pkg/framework/resource/resource.go

```diff
@@ -0,0 +1,68 @@
+package resource
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/helm/pkg/helm"
+
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+)
+
+type ResourceConfig struct {
+	Logger     micrologger.Logger
+	ApprClient *apprclient.Client
+	HelmClient *helmclient.Client
+}
+
+type Resource struct {
+	logger     micrologger.Logger
+	apprClient *apprclient.Client
+	helmClient *helmclient.Client
+}
+
+func New(config ResourceConfig) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.ApprClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ApprClient must not be empty", config)
+	}
+	if config.HelmClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HelmClient must not be empty", config)
+	}
+	c := &Resource{
+		apprClient: config.ApprClient,
+		helmClient: config.HelmClient,
+		logger:     config.Logger,
+	}
+
+	return c, nil
+}
+
+func (r *Resource) InstallResource(name, values, version string, conditions ...func() error) error {
+	chartValuesEnv := os.ExpandEnv(values)
+	chartname := fmt.Sprintf("%s-chart", name)
+
+	tarball, err := r.apprClient.PullChartTarballFromRelease(chartname, version)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	err = r.helmClient.InstallFromTarball(tarball, "giantswarm", helm.ReleaseName(name), helm.ValueOverrides([]byte(chartValuesEnv)), helm.InstallWait(true))
```

Comment:
The org or whatever that is should be configurable. If this is the default then let's default it but having it hard coded is not so cool.

Reply:
This is a weird one: It's a `namespace` - but I could not easily find what the namespace is used for. In the cli there is optional namespace value but this one seems required.

I wanted to hardcode it here and then test what it actually does or which namespace is meant. I couldn't find any documentation which explains if this is the namespace flag similar to the cli or a tiller namespace or something else.

Maybe @fgimenez  has an idea?

Reply:
In our operators we just set it to the namespace of the chart resource in the customObject we watch, doesn't make it much clearer to me.

Reply:
The signature of `helmClient.InstallFromTarball` mimics upstream's `InstallRelease` https://github.com/helm/helm/blob/master/pkg/helm/client.go#L78 in fact it is mostly a proxy method that adds a retry wrapper. Upstream's `InstallRelease` requires a namespace, so does our `helmClient.InstallFromTarball`. 

Regarding the `giantswarm` value, in our current e2e tests all components are installed in this namespace, the same as they are on production. I think it makes sense to parameterize this value and define the default value as a constant. 

Let me know if this makes things more clear?

Reply:
Thanks, I was aware of the proxy to upstream - I kind of don't get the point of the namespace upstream. Charts we install have namespaces set on the individual resources - why do we declare this here again? And it isn't mandatory in helm cli IIRC.

I will add the default like you said, still a tiny bit of confusion.

Reply:
Namespace is in the config now and gets defaulted to `giantswarm`.